src/module/retry.ts

```diff
@@ -14,9 +15,66 @@ export interface PrometheusRetryData extends PrometheusModuleOptions {
 
 export const attachMetrics = (module: Mollitia.Module, options: Mollitia.ModuleOptions): PrometheusRetryMetrics => {
   const metrics = commonMetrics(module, options);
-  // TODO success_without_retry
-  // TODO success_with_retry
-  // TODO failures_without_retry
-  // TODO failures_with_retry
-  return metrics;
-}
+  const labels = {
+    ...options.prometheus?.labels,
+    module: options.prometheus?.name || ''
+  };
+  // Success Without Retries
+  const success_without_retries = new PrometheusCounter(
+    `${options.prometheus?.prefix ? `${options.prometheus?.prefix}_` : ''}success_without_retries`,
+    {
+      description: 'Success Without Retries',
+      labels
+    }
+  );
+  // Success Without Retries
+  const success_with_retries = new PrometheusCounter(
+    `${options.prometheus?.prefix ? `${options.prometheus?.prefix}_` : ''}success_with_retries`,
+    {
+      description: 'Success With Retries',
+      labels
+    }
+  );
+  // Success Without Retries
+  const failures_without_retries = new PrometheusCounter(
+    `${options.prometheus?.prefix ? `${options.prometheus?.prefix}_` : ''}failures_without_retries`,
+    {
+      description: 'Failures Without Retries',
+      labels
+    }
+  );
+  // Success Without Retries
+  const failures_with_retries = new PrometheusCounter(
+    `${options.prometheus?.prefix ? `${options.prometheus?.prefix}_` : ''}failures_with_retries`,
+    {
+      description: 'Failures With Retries',
+      labels
+    }
+  );
+
+  // Handlers
+  module.on('success-without-retry', (circuit: Mollitia.Circuit) => {
+    const metricName = circuit.prometheus.perMethod ? `${circuit.prometheus.name}_${circuit.prometheus.funcName}` : circuit.prometheus.name;
```

Comment:
Maybe we could have a method to compute metricName and use it in the 4 module.on listeners

Reply:
Great catch, will add this to common.ts file.